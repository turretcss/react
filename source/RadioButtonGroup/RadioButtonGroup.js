import React from 'react'

import Button from '../Button'
import ButtonGroup from '../ButtonGroup'

const RadioButtonGroup = ({ options = [], color, activeColor = 'primary', value, onChange, ...props }) => (
  <ButtonGroup {...props}>
    {options.map(option => (
      <Button
        key={option.value}
        label={option.label}
        onClick={() => onChange(option.value)}
        color={value == option.value ? activeColor : color}
      />
    ))}
  </ButtonGroup>
)

export default RadioButtonGroup
