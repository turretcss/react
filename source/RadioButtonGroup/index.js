export { default } from './RadioButtonGroup'
export { default as RadioButtonGroup } from './RadioButtonGroup'
