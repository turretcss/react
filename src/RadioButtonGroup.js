import React from 'react'

import Button from './Button'
import ButtonGroup from './ButtonGroup'

const RadioButtonGroup = ({ options = [], color, activeColor = 'primary', disabled, value, onChange, ...props }) => (
  <ButtonGroup {...props}>
    {options.map(option => (
      <Button
        key={option.value}
        text={option.text}
        disabled={disabled}
        onClick={() => onChange(option.value)}
        color={value == option.value ? activeColor : color}
      />
    ))}
  </ButtonGroup>
)

export default RadioButtonGroup
