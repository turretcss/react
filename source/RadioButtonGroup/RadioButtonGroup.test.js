import React from 'react'
import renderer from 'react-test-renderer'
import { renderToStaticMarkup } from 'react-dom/server'

import RadioButtonGroup from './RadioButtonGroup'

const options = [
  {
    value: null,
    text: 'Auto',
  },
  {
    value: 's',
    text: 'Small',
  },
  {
    value: 'l',
    text: 'Large',
  },
]

describe('RadioButtonGroup', () => {
  it('renders', () => {
    const result = renderToStaticMarkup(<RadioButtonGroup />)
    const expected = '<div class="button-group"></div>'
    expect(result).toEqual(expected)
  })

  it('renders block prop', () => {
    const result = renderToStaticMarkup(<RadioButtonGroup block />)
    const expected = '<div class="button-group button-group-block"></div>'
    expect(result).toEqual(expected)
  })

  it('renders className prop', () => {
    const result = renderToStaticMarkup(<RadioButtonGroup className="custom" />)
    const expected = '<div class="button-group custom"></div>'
    expect(result).toEqual(expected)
  })

  it('renders options as buttons with "Auto" option selected (button-primary)', () => {
    const tree = renderer.create(<RadioButtonGroup options={options} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders options as buttons with "Small" option selected (button-primary)', () => {
    const tree = renderer.create(<RadioButtonGroup value="s" options={options} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('passed button props down to children buttons', () => {
    const tree = renderer.create(<RadioButtonGroup options={options} size="xs" disabled pill square border />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders color and activeColor props', () => {
    const tree = renderer
      .create(<RadioButtonGroup options={options} color="primary" activeColor="secondary" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls onChange when input clicked', () => {
    let newValue = null
    const testInstance = renderer.create(<RadioButtonGroup onChange={value => (newValue = value)} options={options} />)
      .root
    // click button with value "s"
    testInstance.findAllByType('button')[1].props.onClick()
    expect(newValue).toEqual('s')
  })
})
