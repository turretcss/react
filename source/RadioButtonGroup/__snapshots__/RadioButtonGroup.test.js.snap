// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup passed button props down to children buttons 1`] = `
<div
  className="button-group"
>
  <button
    className="button-xs button-primary button-border button-pill button-square"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Auto
  </button>
  <button
    className="button-xs button-border button-pill button-square"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Small
  </button>
  <button
    className="button-xs button-border button-pill button-square"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Large
  </button>
</div>
`;

exports[`RadioButtonGroup renders color and activeColor props 1`] = `
<div
  className="button-group"
>
  <button
    className="button-secondary"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Auto
  </button>
  <button
    className="button-primary"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Small
  </button>
  <button
    className="button-primary"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Large
  </button>
</div>
`;

exports[`RadioButtonGroup renders options as buttons with "Auto" option selected (button-primary) 1`] = `
<div
  className="button-group"
>
  <button
    className="button-primary"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Auto
  </button>
  <button
    className={null}
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Small
  </button>
  <button
    className={null}
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Large
  </button>
</div>
`;

exports[`RadioButtonGroup renders options as buttons with "Small" option selected (button-primary) 1`] = `
<div
  className="button-group"
>
  <button
    className={null}
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Auto
  </button>
  <button
    className="button-primary"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Small
  </button>
  <button
    className={null}
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Large
  </button>
</div>
`;
